custom_components/miele/sensor.py

```diff
@@ -765,18 +763,14 @@ def __init__(
         description: MieleSensorDescription,
     ):
         """Initialize the sensor."""
-        super().__init__(coordinator)
-        self._idx = idx
-        self._ent = ent
-        self.entity_description = description
+        super().__init__(coordinator, idx, ent, description)
         _LOGGER.debug("init sensor %s", ent)
         appl_type = self.coordinator.data[self._ent][self.entity_description.type_key]
         if appl_type == "":
             appl_type = self.coordinator.data[self._ent][
                 "ident|deviceIdentLabel|techType"
             ]
-        self._attr_has_entity_name = True
-        self._attr_unique_id = f"{self.entity_description.key}-{self._ent}"
+        # deviates from MieleEntity
```

Comment:
Is this something that applied to all entities/devices? Then `hw_version ` and `sw_version ` could be moved to `entity.py`

Reply:
Yes, hw_version and sw_version should be moved to entity.py. It works OK today since the effective DeviceInfo class is made up by by OR-ing all properties with the same identifier.